parquet-hadoop/src/test/java/org/apache/parquet/statistics/TestStatistics.java

```diff
@@ -0,0 +1,515 @@
+      final int pageSize = context.pageSize;
+      final int dictionaryPageSize = pageSize;
+      final boolean enableDictionary = context.enableDictionary;
+      final boolean enableValidation = context.enableValidation;
+      ParquetProperties.WriterVersion writerVersion = context.version;
+      CompressionCodecName codec = CompressionCodecName.UNCOMPRESSED;
+
+      ParquetWriter<Group> writer = new ParquetWriter<Group>(context.fsPath, groupWriteSupport, codec, blockSize,
+        pageSize, dictionaryPageSize, enableDictionary, enableValidation, writerVersion, configuration);
+
+      context.write(writer);
+      writer.close();
+
+      context.test();
+      file.delete();
+    }
+  }
+
+  public static abstract class PagingValidator<T extends Comparable<T>> {
+    private static class ComputedStats<T extends Comparable<T>> {
+      int totalNulls = 0;
+      T minimum = null;
+      T maximum = null;
+
+      public void updateStats(T value) {
+        if (value != null) {
+          this.minimum = min(minimum, value);
+          this.maximum = max(maximum, value);
+        } else {
+          this.totalNulls++;
+        }
+      }
+
+      private T min(T lhs, T rhs) {
+        if (lhs == null) return rhs;
+        if (rhs == null) return lhs;
+        return (lhs.compareTo(rhs) < 0) ? lhs : rhs;
+      }
+
+      private T max(T lhs, T rhs) {
+        if (lhs == null) return rhs;
+        if (rhs == null) return lhs;
+        return (lhs.compareTo(rhs) > 0) ? lhs : rhs;
+      }
+
+      public T minimum() { return this.minimum; }
+      public T maximum() { return this.maximum; }
+      public int totalNulls() { return this.totalNulls; }
+    }
+
+    private final ColumnDescriptor columnDescriptor;
+    private final PrimitiveType.PrimitiveTypeName forType;
+
+    public PagingValidator(MessageType schema, String path) {
+      this.columnDescriptor = columnDescriptorByPath(schema, path);
+      this.forType = columnDescriptor.getType();
+    }
+
+    private void validate(PageReadStore store) {
+      PageReader reader = store.getPageReader(columnDescriptor);
+      DataPage page;
+      while ((page = reader.readPage()) != null) {
+        validateStatisticsMatch(getStatisticsFromPageHeader(page), getStatisticsFromPage(page));
+      }
+    }
+
+    private Statistics<T> getStatisticsFromPageHeader(DataPage page) {
+      return page.accept(new DataPage.Visitor<Statistics<T>>() {
+        @Override
+        public Statistics<T> visit(DataPageV1 dataPageV1) {
+          return (Statistics<T>)dataPageV1.getStatistics();
+        }
+
+        @Override
+        public Statistics<T> visit(DataPageV2 dataPageV2) {
+          return (Statistics<T>)dataPageV2.getStatistics();
+        }
+      });
+    }
+
+    private ComputedStats<T> getStatisticsFromPage(final DataPage page) {
+      return page.accept(new DataPage.Visitor<ComputedStats<T>>() {
+        @Override
+        public ComputedStats<T> visit(DataPageV1 dataPageV1) {
```

Comment:
Why is this computing stats instead of getting the stats object written with the page?